MekHQ/src/mekhq/gui/baseComponents/AbstractMHQButtonDialog.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.gui.baseComponents;
+
+import megamek.client.ui.baseComponents.AbstractButtonDialog;
+import megamek.common.util.EncodeControl;
+import mekhq.MekHQ;
+
+import javax.swing.*;
+import java.util.ResourceBundle;
+
+/**
+ * This is the Base Dialog for a dialog with buttons in MegaMek. It extends Base Dialog, and adds a
+ * button panel with base Ok and Cancel buttons. It also includes an enum tracker for the result of
+ * the dialog.
+ *
+ * Inheriting classes must call initialize() in their constructors and override createCenterPane()
+ *
+ * The resources associated with this dialog need to contain at least the following keys:
+ * - "Ok.text" -> text for the ok button
+ * - "Ok.toolTipText" -> toolTipText for the ok button
+ * - "Cancel.text" -> text for the cancel button
+ * - "Cancel.toolTipText" -> toolTipText for the cancel button
+ */
+public abstract class AbstractMHQButtonDialog extends AbstractButtonDialog {

```

Comment:
Gotta ask, what's different about a MekHQ button from a MegaMek button?

Reply:
Solely tracking on MHQ's preferencez instead of MM's.

Otherwise the dialogs are identical.